lib/flight_scheduler/application.rb

```diff
@@ -64,5 +68,22 @@ def config
     def root
       @root ||= Pathname.new(__dir__).join('../../').expand_path
     end
+
+    def init_timer_tasks
```

Comment:
You're quite right about the name not being terribly good.  I imagine that this periodic thingy will expand to include other thingies.  Perhaps allocating jobs and perhaps persisting the job registry.  So its purpose probably isn't updating the registry, but running periodic thingies. 

A very important distinction between `TimerTask` and the block you wrote is that `TimerTask` is robust in the face of exceptions.  In particular, an exception thrown by its block will not prevent the task running again when due.  That is the primary reason that I chose it.

A benefit of the consistent interval/timeout is that it gives a guarantee that the task will not be running for at least half the time.  For processor heavy background threads this is a really nice guarantee.  I'm not sure that this is a processor heavy background thread however. I'm also not sure how fast `remove_old_jobs` is going to be.  It will involve deleting jobs from the file system and perhaps eventually writing out accounting information.

On that note, I've just realised that both `BatchScript#write` and `BatchScript#cleanup` are blocking operations.  These do not play nicely with our `Async` usage.  They will need to be fixed, especially if this background thread is to use the main `Async` fiber.

You make a good point about `JobRegistry` and the thread being killed due to a timeout.  The read/write lock will cope just fine.  However, as currently written we risk deleting an array job and leaving its tasks behind.  We should be robust here regardless of the concurrency mechanism.

Reply:
I've made a few changes to `JobRegistry` and I've convinced myself that it is robust even if the periodic processor is timed out whilst it is running.  `JobRegistry` is thread safe so I can't see any issues with it being accessed by a background thread.